test/functional/helpers/optIn/createMockOptIn.js

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { ClientFunction } from "testcafe";
+
+export default ClientFunction(approved => {
+  window.adobe = {
+    optIn: {
+      fetchPermissions(callback) {
+        setTimeout(() => {
+          callback();
+        }, 500);
```

Comment:
Is the `500` important here versus, say `0`? Is there a way we can make it not important while still providing a reasonable test? I'm just always hesitant to add delays in automated tests because they start to add up fast.

Reply:
I just want the optIn to be pending for a little while. I could change it to explicitly calling it callback but I'd rather reduce it (which I think is also fine).

Reply:
Unless it will actually influence the result of the test, I'd put it on a timeout of `0`.

Reply:
@Aaronius , regarding @wgamboa 's comment. That makes sense that we are looking at optin separately in Identity.